lib/macroni/Conversion/Kernel/KernelRewriters.cpp

```diff
@@ -151,8 +176,144 @@ mlir::LogicalResult rewrite_rcu_dereference(macroni::MacroExpansion exp,
   }
 
   auto p_clone = rewriter.clone(*p);
-  rewriter.replaceOpWithNewOp<RCUDereference>(exp, exp.getType(0),
-                                              p_clone->getResult(0));
+  if (has_name(exp, "rcu_dereference")) {
+    rewriter.replaceOpWithNewOp<RCUDereference>(exp, exp.getType(0),
+                                                p_clone->getResult(0));
+  } else if (has_name(exp, "rcu_dereference_bh")) {
+    rewriter.replaceOpWithNewOp<RCUDereferenceBH>(exp, exp.getType(0),
+                                                  p_clone->getResult(0));
+  } else if (has_name(exp, "rcu_dereference_sched")) {
+    rewriter.replaceOpWithNewOp<RCUDereferenceSched>(exp, exp.getType(0),
+                                                     p_clone->getResult(0));
```

Comment:
You can fetch the mnemonic from te class - then all of these branches can be implemented by one helper that is supplied the type?